packages/stencil/components/pagination/src/components/osds-pagination/osds-pagination.tsx

```diff
@@ -0,0 +1,344 @@
+      }
+
+      if (endIndex < totalPages) {
+        pageList[totalPages - 1].active = true;
+      }
+    }
+
+    return (
+      <Host
+        {...{
+          class: `${disabled ? 'disabled' : ''}${this.hasError() ? ' ods-error' : ''}`,
+          onFocus: this.onFocus.bind(this),
+          onBlur: this.onBlur.bind(this),
+          pageIndex: this.disabled ? -1 : this.pageindex,
+          totalPages: this.totalPages,
+          current: this.current,
+        }}
+      >
+        <div>
+          <ul>
+            <li>
+              <osds-button
+                contrasted={true}
+                color={OdsThemeColorIntent.primary}
+                disabled={this.pageindex > 1 ? false : true}
+                onKeyDown={(event: any) => {
+                  if (this.pageindex > 1 && !disabled) this.onKeyPress(event, Number(this.pageindex) - 1);
+                }}
+                onClick={() => {
+                  this.setPageIndex(Number(this.pageindex) - 1);
+                }}
+                size={OdsButtonSize.sm}
+              >
+                <osds-icon size={OdsIconSize.sm} name={OdsIconName.CHEVRON_LEFT} color={OdsThemeColorIntent.primary}></osds-icon>
+              </osds-button>
+            </li>
+            {pageList.map(page => {
+              if (page.active)
+                return (
+                  <span>
+                    {pageList.length > 6 && pageList.length - this.pageindex > 3 && page.id == pageList.length && (
+                      <osds-icon
+                        size={OdsIconSize.sm}
+                        color={OdsThemeColorIntent.primary}
+                        name={OdsIconName.ELLIPSIS}
+                        style={{ paddingBottom: '10px', cursor: 'not-allowed' }}
+                      ></osds-icon>
+                    )}
+                    <li>
+                      <osds-button
+                        class={this.pageindex == page.id ? 'selectedpage' : ''}
+                        contrasted={this.pageindex == page.id ? false : true}
+                        color={OdsThemeColorIntent.primary}
+                        size={OdsButtonSize.sm}
+                        onKeyDown={(event: any) => {
+                          if (!disabled) this.onKeyPress(event, Number(page.id));
+                        }}
+                        onClick={() => {
+                          this.setPageIndex(Number(page.id));
+                        }}
+                      >
+                        {page.id}
+                      </osds-button>
+                    </li>
+                    {pageList.length > 6 && this.pageindex > 4 && page.id == 1 && (
+                      <li>
+                        <osds-icon
+                          size={OdsIconSize.sm}
+                          color={OdsThemeColorIntent.text}
+                          name={OdsIconName.ELLIPSIS}
+                          style={{ paddingBottom: '10px', cursor: 'not-allowed' }}
```

Comment:
Please don't add style here but in the scss file with selector osds-icon {}
paddingBottom is not necessary here, on the sketch we can see ellipsis icon should be kept at the bottom

Reply:
osds-icon is also used for the arrows, so we need to use a class specific for the ellipsis to not impact the arrows that are also osds-icon